Sources/SwiftProcessing/Core/Sketch.swift

```diff
@@ -167,27 +177,38 @@ import SceneKit
         updateTimes()
         self.rect = rect
         sketchDelegate?.draw()
-        
+
         postDraw3D()
-        
+
         updateTouches()
     }
-    
+
     private func updateTimes() {
         frameCount =  frameCount + 1
         let newTime = CGFloat(CACurrentMediaTime())
         deltaTime = newTime - lastTime
         lastTime = newTime
     }
-    
+
     open func push() {
         context?.saveGState()
         settingsStack.push(settings: settings)
+
+        let rootTransformationNode = self.currentTransformationNode
```

Comment:
I doubt it'd be a significant performance bump, but I think it makes sense to wrap this with an if statement checking to see if 3D mode is enabled